### PR TITLE
improvement(k8s): add K8S 1.27 support

### PR DESCRIPTION
**Pull request by vponomaryov**
Following main things are done here:
- The hydra image is updated with all the required stuff for the K8S 1.27 support
- Update the SCT parts which depend on the new approach working with GKE clusters.
    
Changes:
- Update the K8S server version to be `1.27` for EKS and GKE because these
  are the most recent ones in those K8S cloud platforms.
- Update `kubectl` and `eksctl` binaries versions to match the server version.
- Install the `google-cloud-sdk-gke-gcloud-auth-plugin`
  package which is required for the GKE auth starting from the `K8S 1.26`.
- Install the `google-cloud-sdk` package which is required by
  the `google-cloud-sdk-gke-gcloud-auth-plugin` one.
  It adds the `gcloud` binary which is used for managing GKE clusters.
- Install `helm` package and use it from the `hydra` because it uses
  `kubectl` which uses `gke-gcloud-auth-plugin` plugin for auth.
- Stop using `token update thread` for GKE because this way of auth is not supported in the K8S 1.26+ versions.
  The `gke-gcloud-auth-plugin` plugin will be used for the auth.

Also, switch GKE to the dynamic PV provisioner as the only working solution after the switch.
The old `static PV provisioner` conflicts with the new K8S versions which leads to some K8S Scylla nodes failures.

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: docker/env/Dockerfile [hunk 1]
```diff
@@ -19,8 +19,8 @@ RUN mkdir /build && \
     pip3 install -r requirements.txt --root=/build --prefix=./ --ignore-installed --no-warn-script-location
 
 FROM python:$PYTHON_IMAGE_TAG
-ARG KUBECTL_VERSION=1.23.14
-ARG EKSCTL_VERSION=0.124.0
+ARG KUBECTL_VERSION=1.27.3
```
Comment (fruch) [inline docker/env/Dockerfile:22]: lets build a docker with this

**Comment by vponomaryov** [comment]
The PR got much bigger because of GKE and it's requirements for K8S 1.26+
For details see the commit or PR description.

**Comment by vponomaryov** [comment]
Tested here: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/valerii/job/vp-longevity-scylla-operator-3h-gke/99/consoleFull
All changed parts already worked:
- gcloud commands
- helm commands
- kubectl commands

**Comment by vponomaryov** [comment]
Ok, testing showed that our static provisioner logic breaks some Scylla K8S node if we use K8S 1.27
So, we must use dynamic provisioner in GKE with latest K8S versions.
Will add additional commit here instead of the separate PR as was planned to.

**Comment by vponomaryov** [comment]
This one is ready for review.
Tested here: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/valerii/job/vp-longevity-scylla-operator-3h-gke/100/consoleFull

**Review by fruch — APPROVED** [review]
LGTM

**Review by soyacz — APPROVED** [review]

**Pull request merged**